pkg/webhooks/kafkacluster_validator.go

```diff
@@ -60,14 +65,24 @@ func (s KafkaClusterValidator) ValidateDelete(ctx context.Context, obj runtime.O
 }
 
 // checkBrokerStorageRemoval checks if there is any broker storage which has been removed. If yes, admission will be rejected
-func checkBrokerStorageRemoval(kafkaClusterSpecOld, kafkaClusterSpecNew *banzaicloudv1beta1.KafkaClusterSpec) *field.Error {
+func checkBrokerStorageRemoval(kafkaClusterSpecOld, kafkaClusterSpecNew *banzaicloudv1beta1.KafkaClusterSpec) (*field.Error, error) {
 	for j := range kafkaClusterSpecOld.Brokers {
 		brokerOld := &kafkaClusterSpecOld.Brokers[j]
 		for k := range kafkaClusterSpecNew.Brokers {
 			brokerNew := &kafkaClusterSpecNew.Brokers[k]
 			if brokerOld.Id == brokerNew.Id {
-				brokerConfigsOld, _ := brokerOld.GetBrokerConfig(*kafkaClusterSpecOld)
-				brokerConfigsNew, _ := brokerNew.GetBrokerConfig(*kafkaClusterSpecNew)
+				brokerConfigsOld, err := brokerOld.GetBrokerConfig(*kafkaClusterSpecOld)
+				if err != nil {
+					return nil, err
+				}
+				// checking broukerConfigGroup existence
+				if _, exists := kafkaClusterSpecNew.BrokerConfigGroups[brokerNew.BrokerConfigGroup]; !exists {
```

Comment:
check for `brokerNew.BrokerConfigGroup != ""` as well as it's not mandatory to use broker config group

Reply:
Good catch :)